### DISCUSSION TITLE
Dynamic/flexible input.harbor for *any* connected mount

### DISCUSSION THREAD

**Discussion by tibus-hoti** [savonet/liquidsoap#4069] [Q&A]
For a fixed input mount,  I can create fixed outputs. e.g.

```
# Set up a harbor input
input = input.harbor("stream", port=8000, password="hackme")

# Output to an Icecast server
output.icecast(%mp3,
  host = "localhost", port = 8001,
  password = "hackme", mount = "/stream-128",
  fallible=true, input)

output.icecast(
  %ffmpeg(format="mp3", %audio(codec="libmp3lame", b="64k")),
  mount="/stream-64",
  host="localhost", port=8001, password="hackme",
  fallible=true, input)
```

But I would  like to accept ANY mount name that is connected to input.habor and transcode a lower bit rate version of any that connect, preferably taking the input mount name to use in the output mount name, with the bitrate tacked on eg.  (input mount name + "-64"). Bonus points if I can detect the input stream type (mp3/aac)  and apply that to the outputs too, rather than defining it all statically. So, if the input mount connected is aac, apply the outputs as above, but relevant to aac.

Thanks in advance for  any assistance.

**Comment by tibus-hoti** [comment]
Replying to my own post, but I managed to figure this out and can now dynamically create inputs via telnet, connect to the new input and have it then dynamically transcode my various bitrates!  Just need  a basic web UI in place  of telnet and I am good to go.

**Reply by vitoyucepi** [reply]
Hi @tibus-hoti,
It's good to know that you find an answer yourself.
I haven't seen any method to create `input.harbor` with wildcard mount. If I were you, I would make an HTTP API using liquidsoap, instead of telnet, that will allow creating and deleting mounts.

**Reply by tibus-hoti** [reply]
Thanks.  I am using telnet just to test, HTTP implementation is next, potentially storing the additions, so they are created on service restart. 

Basically I have a default mount "default" that starts up the harbor port, but then I can create new inputs via the telnet commands, which it then allows encoder to connect and also creates the various bitrates output dynamically. Its working well, in test form. I can post my solution in case it might be useful to some as a starter.
